packages/gitbook/src/components/DocumentView/Expandable/Details.tsx

```diff
@@ -0,0 +1,98 @@
+'use client';
+
+import { useParams } from 'next/navigation';
+import React from 'react';
+
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+function useHash() {
+    const params = useParams();
+    const [hash, setHash] = React.useState<string>(global.location?.hash?.slice(1));
+    React.useEffect(() => {
+        function updateHash() {
+            setHash(global.location?.hash?.slice(1));
+        }
+        global.addEventListener('hashchange', updateHash);
+        updateHash();
+        return () => global.removeEventListener('hashchange', updateHash);
+    }, [params]);
+    return hash;
+}
+
+/**
+ * Details component rendered on client so it can expand dependent on url hash changes.
+ */
+export function Details(props: {
+    children: React.ReactNode;
+    id: string;
+    contentIds?: string[];
+    open?: boolean;
+    className?: ClassValue;
+}) {
+    const { children, id, open, className } = props;
+
+    const detailsRef = React.useRef<HTMLDetailsElement>(null);
+
+    const [anchorElement, setAnchorElement] = React.useState<Element | null | undefined>();
+
+    const hash = useHash();
+    React.useEffect(() => {
+        if (!hash) {
+            return;
+        }
+        const descendant =
+            hash === id ? detailsRef.current : detailsRef.current?.querySelector(`#${hash}`);
+        setAnchorElement(descendant);
+    }, [hash, id]);
+
+    React.useLayoutEffect(() => {
+        if (anchorElement) {
+            anchorElement.scrollIntoView({
```

Comment:
I just noticed we also have `useScrollToHash`, I'm slightly concerned of the conflict between the 2 into trying to scroll.